File: all/abc312/abc312_a/abc312_a.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  string s;
  cin >> s;

  vector<string> ptn = {"ACE", "BDF", "CEG", "DFA", "EGB", "FAC", "GBD"};
  for(auto t : ptn) {
    if(s == t) {
      cout << "Yes";
      return 0;
    }
  }

  cout << "No";
  return 0;
}

File: all/abc312/abc312_a/abc312_a.cs
class A {
  static void Main() {
    string s = Console.ReadLine();
    string[] ptn = {"ACE", "BDF", "CEG", "DFA", "EGB", "FAC", "GBD"};

    foreach(string t in ptn)
    {
      if(s == t)
      {
        Console.WriteLine("Yes");
        return;
      }
    }

    Console.WriteLine("No");
  }
}
